src/lib/wsman-filter.c

```diff
@@ -357,7 +357,8 @@ filter_t * filter_deserialize(WsXmlNodeH node, const char *ns)
 {
 	char *dialect = NULL;
 	int properNum = 0;
-	int i = 0;
+	unsigned int i = 0;
```

Comment:
the declaration (and initialization) of `i` should be moved to the block of actual use (the `if(strcmp(filter->dialect , WSM_SELECTOR_FILTER_DIALECT) == 0) {` one)